addons/sale/models/sale.py

```diff
@@ -202,6 +202,13 @@ def _get_payment_type(self):
         ('confirmation_date_conditional_required', "CHECK( (state IN ('sale', 'done') AND confirmation_date IS NOT NULL) OR state NOT IN ('sale', 'done') )", "A confirmed sales order requires a confirmation date."),
     ]
 
+    @api.multi
+    @api.constrains('company_id')
+    def _check_product_from_own_company(self):
+        for line in self.mapped('order_line'):
+            if line.product_id.company_id and line.product_id.company_id != self.company_id:
+                raise ValidationError(_("Your quotation belongs to company %s and cannot contains products restricted to another company (%s belonging to %s). You should use another product or share the product with all companies." % (self.company_id.name, line.product_id.name, line.product_id.company_id.name)))
```

Comment:
`product_id.name` --> `product_id.name_get()[0][1]`

Reply:
`product_id.name_get()[0][1]` --> `product_id.display_name`

Reply:
Oh yes, (Missed it)